include/asterisk/res_pjsip_body_generator_types.h

```diff
@@ -74,4 +74,30 @@ struct ast_sip_message_accumulator {
 	char message_account[PJSIP_MAX_URL_SIZE];
 };
 
+/*!
+ * \brief Synchronization data used for as-feature-event XML bodies
+ *
+ * This is used for x-as-feature-event+xml bodies.
+ */
+struct ast_sip_device_feature_sync_data {
+	/* Device ID */
+	char deviceid[64];
```

Comment:
It would hold the device name (channel name before the -).
Where is the #define for the max device name length? I'd been looking for that for a long time and could never find one. I know you and Sean were discussing the one for channel name length, but that would be longer than necessary here.

Reply:
This "struct ast_sip_device_feature_sync_data" should be the sorcery object. For example
struct ast_sip_device_feature_sync_data {
  /*! Sorcery object details, the id is the ENDPOINT name */
  SORCERY_OBJECT(details);
  AST_DECLARE_STRING_FIELDS(
    AST_STRING_FIELD(fwd_exten_always);
    AST_STRING_FIELD(fwd_exten_busy);
    AST_STRING_FIELD(fwd_exten_noanswer);
  );
  unsigned int dnd;
  unsigned int ring_count;
};

Reply:
Sorry, forget deviceid
AST_STRING_FIELD(deviceid);